zulip/integrations/bridge_with_matrix/matrix_bridge.py

```diff
@@ -26,27 +26,33 @@
 class Bridge_ConfigException(Exception):
     pass
 
+class Bridge_MatrixException(Exception):
+    pass
+
+class Bridge_ZulipException(Exception):
+    pass
+
 def matrix_login(matrix_client: Any, matrix_config: Dict[str, Any]) -> None:
     try:
         matrix_client.login_with_password(matrix_config["username"],
                                           matrix_config["password"])
     except MatrixRequestError as e:
         if e.code == 403:
-            sys.exit("Bad username or password.")
+            raise Bridge_MatrixException("Bad username or password.")
         else:
-            sys.exit("Check if your server details are correct.")
+            raise Bridge_MatrixException("Check if your server details are correct.")
     except MissingSchema as e:
-        sys.exit("Bad URL format.")
+        raise Bridge_MatrixException("Bad URL format.")
 
 def matrix_join_room(matrix_client: Any, matrix_config: Dict[str, Any]) -> Any:
     try:
         room = matrix_client.join_room(matrix_config["room_id"])
         return room
     except MatrixRequestError as e:
         if e.code == 403:
-            sys.exit("Room ID/Alias in the wrong format")
+            raise Bridge_MatrixException("Room ID/Alias in the wrong format")
         else:
-            sys.exit("Couldn't find room.")
+            raise Bridge_MatrixException("Couldn't find room.")
```

Comment:
This isn't new to your code, but I'm not a fan of these else clauses for non-403 errors assuming what the error is; can't we just pass through whatever the text of the `MatrixRequestError` was?

Reply:
This seems reasonable to some degree, though if we have a concrete idea that the error is something specific, I'm wondering if it makes sense to interpret it before wrapping it in the different `Exception`? Or maybe chaining with something like:
`Bridge_MatrixException("Couldn't find room. "+ str(exception))` ?

@rheaparekh You added this error handling; could you comment?

Reply:
I had been following the sample codes for the official `matrix_client` for this, but the sample codes aren't consistent which each other as well. It will be better to pass just the text of the `Exception` without chaining it.

Reply:
Sample code: https://github.com/matrix-org/matrix-python-sdk/tree/master/samples

Reply:
OK, sounds like the library is a bit sloppy.  Let's hold off on trying to clean this up until we've done the rest.